### PR TITLE
Create an Authorize middleware

### PR DESCRIPTION
**Pull request by JosephSilber**
Can be used with a simple ability:

```php
Route::group(['middleware' => 'can:view-dashboard'], function () {
    // Dashboard routes...
});
```

Can be used on a single model:

```php
Route::get('users/{user}', 'UserController@show')->middleware('can:view,user');
```

It will pull the `$user` model from the route.

Finally, it can be used on a model type:

```php
Route::get('users', 'UserController@index')->middleware('can:view,'.User::class);
```

### CODE REVIEW

File: app/Http/Middleware/Authorize.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    protected function getGateArguments($request, $model)
+    {
+        // If there's no model, we'll pass an empty array to the gate. If it
+        // looks like a FQCN of a model, we'll send it to the gate as is.
+        // Otherwise, we'll resolve the Eloquent model from the route.
+        if (is_null($model)) {
+            return [];
+        }
+
+        if (strpos($model, '\\') !== false) {
```
Comment (sileence) [inline app/Http/Middleware/Authorize.php]: this could use a comment (if this is a class name...)

Comment (JosephSilber) [inline app/Http/Middleware/Authorize.php]: I think the first comment is enough.

Comment (sileence) [inline app/Http/Middleware/Authorize.php]: maybe leaving the first comment but using else if below? To make this one block.

**Comment by JosephSilber** [comment]
Closing this in favor of [having the middleware in core](https://github.com/laravel/framework/pull/12913).

**Pull request closed without merging**